File: ec2-cityapp.py
import mysql.connector
import boto3
import json 

REGION="ap-southeast-1"

client = boto3.client('secretsmanager',region_name=REGION)
response = client.get_secret_value(SecretId='jtan-rds-cityapp-2')
database_secrets = json.loads(response['SecretString'])

USER=database_secrets['username']
PASSWORD=database_secrets['password']
HOST=database_secrets['host']
DBNAME=database_secrets['dbname']

db = mysql.connector.connect(
  host=HOST,
  user=USER,
  password=PASSWORD,
  database=DBNAME
)

cursor = db.cursor()

cursor.execute("SELECT city.Name as City,country.name as Country,city.District,city.Population FROM city,country WHERE city.CountryCode = country.Code ORDER BY RAND() LIMIT 0,1")

result = cursor.fetchall()

for row in result:
  print(row)

File: ec2-cityapp.js
const secretRegion = 'ap-southeast-1'
const secretName = 'jtan-rds-cityapp-2'
const {SecretsManagerClient,GetSecretValueCommand} = require('@aws-sdk/client-secrets-manager')
const client = new SecretsManagerClient({region: secretRegion})

const mysql = require('mysql2')
const query = 'SELECT city.Name as City,country.name as Country,city.District,city.Population FROM city,country WHERE city.CountryCode = country.Code ORDER BY RAND() LIMIT 0,1'

async function getSecret(){
  const response = await client.send(
    new GetSecretValueCommand({
      SecretId: secretName,
      VersionStage: 'AWSCURRENT'
    })
  )
  const result = JSON.parse(response.SecretString)
  return result
}

getSecret().then(secret => {
  const connectionConfig = {
    host: secret['host'],
    user: secret['username'],
    password: secret['password'],
    database: secret['dbname']
  }
  const connection = mysql.createConnection(connectionConfig)
  connection.connect(function(err){
    if (err) throw err
    connection.query(query, function(err,results,fields){
      if (err) throw err
      for(row of results){
        console.log(row)
      }
    })
    connection.end()
  })
})
